Automattic/sensei
# New WooCommerce settings section

Fixes #4758

### Changes proposed in this Pull Request
- Add new WooCommerce section to the Settings Page
- Style the page according to the screen dimensions

### About the design 
- Design: [Figma](https://www.figma.com/file/o9Mn6O9QYab4bRMkjdbwHL/Sensei-Upgrade-Triggers?node-id=259%3A1576 
)
- There are two options of the design for smaller screens I took the one where we show the image 

### Testing instructions
- Feature is by default enabled
- Open the Sensei LMS -> Settings 
1. You should see new tab called **'WooCommerce'**
2. When you open it try clicking on the _**Upgrade to Sensei Pro**_ button it should redirect you to this page: https://senseilms.com/pricing/ (the page doesn't exist yet)
3. When resizing screens you should see following UI-s

@[Luchadores](https://github.com/Luchadores) can you check the designs in screenshot, since I've added adjustable padding as we agreed 

| Mobile View < 440| Tablet/Ipad View | Smaller desktop view < 1100 | Normal desktop view | Very big screen ( > 1500) |
| -- | -- | -- | -- | -- |
| <img width="331" alt="image" src="https://user-images.githubusercontent.com/7208249/153568805-c5a1c155-223c-42a7-a361-e8c25a27657d.png"> | <img width="479" alt="image" src="https://user-images.githubusercontent.com/7208249/153568864-3583c371-f3c2-4884-a5ff-76f5f4bc538c.png"> | <img width="1024" alt="image" src="https://user-images.githubusercontent.com/7208249/153568928-cda7e0bf-0b55-4977-8b30-184c06a23c8c.png"> | <img width="1512" alt="image" src="https://user-images.githubusercontent.com/7208249/153568969-fa809c0a-c676-4bda-b149-e5536dc45b53.png"> | <img width="1512" alt="image" src="https://user-images.githubusercontent.com/7208249/153569253-5b1ded69-60b6-497a-a009-e0dc8ecdec23.png"> |



### New/Updated Hooks
* Added hook: **sensei_settings_woocommerce_hide** so that it can be used to hide WooCommerce section if not needed  

### For discussion: 
1. Why I didn't make this a notice but section instead: 
- It is too different from the notice: doesn't have to dismiss option, doesn't have the same elements
- We only need it on one page
- Not displayed on the same place where we display the notice (which is on top of everything) 

2. Why am I hiding elements with CSS and not in settings.js
-  was buggy and worked well with CSS

3. Why am I shifting images in settings.js: couldn't think of better way
